WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -1161,13 +1161,29 @@ private extension ProductsViewController {
     }
 
     func configureLeftBarBarButtomItemAsScanningButtonIfApplicable() {
-        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.scanToUpdateInventory),
-              UIImagePickerController.isSourceTypeAvailable(.camera) else {
-            navigationItem.leftBarButtonItem = nil
-            return
+        // The feature breaks if the Square plugin is active, since modifies inventory management logic
+        // If the plugin is active, we'll hide the inventory scanner button
+        // More details: https://wp.me/pdfdoF-2Nq
+        isPluginActive(SitePlugin.SupportedPlugin.square) { result in
```

Comment:
Can we move this logic to the view model `ProductListViewModel`, and add tests? we could expose a function such as `scanToUpdateInventoryButtonShouldBeVisible`, that checks internally whether the square plugin is active.

Reply:
Sure thing, done on https://github.com/woocommerce/woocommerce-ios/pull/11420/commits/f2c4c3b296e3856df447a45fd08adb6c96c5a3c5. 

Regarding tests, I've added tests for feature-flag check on https://github.com/woocommerce/woocommerce-ios/pull/11420/commits/bb89fce28eae354ce51290be65c55a0853953906, but I think adding additional tests to check about the plugin will add quite an overhead, since we also may need to import and inject storage into the model. Do you think these tests are enough? Another option I'm thinking could be to create some sort of "checker service" with a protocol, and then inject a mock to test its behaviour, but again seems a bit of overhead for logic that is tested elsewhere. wdyt?